tests/protocol/test_wire.py

```diff
@@ -492,24 +430,20 @@ def test_get_ext_conf_without_uri(self, *args):
             self.assertIsNone(ext_conf.status_upload_blob_type)
             self.assertIsNone(ext_conf.artifacts_profile_blob)
 
-    def test_get_ext_conf_with_uri(self, *args):
-        mock_data = mockwiredata.WireProtocolData(mockwiredata.DATA_FILE)
-
-        with patch("azurelinuxagent.common.utils.restutil.http_get", side_effect=mock_data.mock_http_get):
-            with patch("azurelinuxagent.common.protocol.wire.CryptUtil", side_effect=mock_data.mock_crypt_util):
-                protocol = WireProtocol(WIRESERVER_URL)
-                # This call would populate the ext_conf with the test data and also mock the required certs
-                protocol.detect()
-                wire_protocol_client = protocol.client
-                ext_conf = wire_protocol_client.get_ext_conf()
-
-                self.assertEqual(1, len(ext_conf.ext_handlers.extHandlers))
-                self.assertEqual(2, len(ext_conf.vmagent_manifests.vmAgentManifests))
-                self.assertEqual("https://test.blob.core.windows.net/vhds/test-cs12.test-cs12.test-cs12.status?"
-                                 "sr=b&sp=rw&se=9999-01-01&sk=key1&sv=2014-02-14&sig=hfRh7gzUE7sUtYwke78IOlZOrTRCYvkec4hGZ9zZzXo",
-                                 ext_conf.status_upload_blob)
-                self.assertEqual("BlockBlob", ext_conf.status_upload_blob_type)
-                self.assertEqual(None, ext_conf.artifacts_profile_blob)
+    def test_get_ext_conf_with_uri(self):
+        with mock_wire_protocol(mockwiredata.DATA_FILE) as protocol:
+            protocol.detect()
```

Comment:
mock_wire_protocol returns a fully initialized protocol. No need to call detect() again

Reply:
Tnx, leftover from old test.